cirq-core/cirq/devices/grid_qubit.py

```diff
@@ -178,22 +171,33 @@ class GridQid(_BaseGridQid):
     cirq.GridQid(5, 4, dimension=2)
     """
 
-    def __init__(self, row: int, col: int, *, dimension: int) -> None:
-        """Initializes a grid qid at the given row, col coordinate
+    _cache = weakref.WeakValueDictionary[Tuple[int, int, int], 'cirq.GridQid']()
```

Comment:
A line comment explaining this would be appreciated.

Reply:
Added.